r/src/array.cpp

```diff
@@ -288,4 +289,10 @@ bool Array__Same(const std::shared_ptr<arrow::Array>& x,
                  const std::shared_ptr<arrow::Array>& y) {
   return x.get() == y.get();
 }
+
+// [[arrow::export]]
+int64_t Array__ReferencedBufferSize(const std::shared_ptr<arrow::Array>& x) {
+  auto array = arrow::internal::checked_cast<const arrow::Array*>(x.get());
+  return ValueOrStop(arrow::util::ReferencedBufferSize(*array));
```

Comment:
The checked_cast is pointless here, the type is the same left and right.
```suggestion
  return ValueOrStop(arrow::util::ReferencedBufferSize(*x));
```